programs/uxd/src/instructions/edit_controller.rs

```diff
@@ -41,32 +50,60 @@ pub(crate) fn handler(ctx: Context<EditController>, fields: &EditControllerField
         let identity_depository_weight_bps = depositories_weight_bps.identity_depository_weight_bps;
-        let mercurial_vault_depository_0_weight_bps =
-            depositories_weight_bps.mercurial_vault_depository_0_weight_bps;
-        let credix_lp_depository_0_weight_bps =
-            depositories_weight_bps.credix_lp_depository_0_weight_bps;
+        let mercurial_vault_depository_weight_bps =
+            depositories_weight_bps.mercurial_vault_depository_weight_bps;
+        let credix_lp_depository_weight_bps =
+            depositories_weight_bps.credix_lp_depository_weight_bps;
         msg!(
             "[edit_controller] identity_depository_weight_bps {}",
             identity_depository_weight_bps
         );
         msg!(
-            "[edit_controller] mercurial_vault_depository_0_weight_bps {}",
-            mercurial_vault_depository_0_weight_bps
+            "[edit_controller] mercurial_vault_depository_weight_bps {}",
+            mercurial_vault_depository_weight_bps
         );
         msg!(
-            "[edit_controller] credix_lp_depository_0_weight_bps {}",
-            credix_lp_depository_0_weight_bps
+            "[edit_controller] credix_lp_depository_weight_bps {}",
+            credix_lp_depository_weight_bps
         );
         controller.identity_depository_weight_bps = identity_depository_weight_bps;
-        controller.mercurial_vault_depository_0_weight_bps =
-            mercurial_vault_depository_0_weight_bps;
-        controller.credix_lp_depository_0_weight_bps = credix_lp_depository_0_weight_bps;
+        controller.mercurial_vault_depository_weight_bps = mercurial_vault_depository_weight_bps;
+        controller.credix_lp_depository_weight_bps = credix_lp_depository_weight_bps;
         emit!(SetDepositoriesWeightBps {
             version: controller.version,
             controller: ctx.accounts.controller.key(),
             identity_depository_weight_bps,
-            mercurial_vault_depository_0_weight_bps,
-            credix_lp_depository_0_weight_bps,
+            mercurial_vault_depository_weight_bps,
+            credix_lp_depository_weight_bps,
+        });
+    }
+
+    // Optionally edit all depositories addresses
+    if let Some(depositories) = fields.depositories {
+        let identity_depository = depositories.identity_depository;
+        let mercurial_vault_depository = depositories.mercurial_vault_depository;
+        let credix_lp_depository = depositories.credix_lp_depository;
+        msg!(
+            "[edit_controller] identity_depository {}",
+            identity_depository
+        );
+        msg!(
+            "[edit_controller] mercurial_vault_depository {}",
+            mercurial_vault_depository
+        );
+        msg!(
+            "[edit_controller] credix_lp_depository {}",
+            credix_lp_depository
+        );
+        controller.identity_depository = identity_depository;
```

Comment:
Should we check the provided pubkey is a part of registered controller depositories?

Reply:
As long as we keep them.. yes probably for consistency, although it's a useless check in absolute. We should refactor the concept of registered depository out

Reply:
I think for now we can keep it that way:
- router mint/redeem check for EXACTLY `router_depositories` matches
- regular mint/redeem depository specifics check for "contained in" `registered_depositories`

I think it makes the most sense as they semantically have different purposes, but long term we should remove `registered_depositories` completely IMO, a bit too much for right now in this PR

Reply:
My concern was what happens if you put an address that doesn't exist?

Reply:
Yeah I agree.

So the possible cases are:
- 1) wrong router_depositories content -> the router stops working and single mint/redeem keeps working
- 2) wrong registered_depositories content -> everything stops working

I think long term we can get rid of N2